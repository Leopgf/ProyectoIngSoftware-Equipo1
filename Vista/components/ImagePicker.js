import React, { useState, useEffect } from 'react';
import { Image, View, Platform, Dimensions } from 'react-native';
import * as ImagePicker from 'expo-image-picker';
import { StyleSheet } from 'react-native';
import { Button } from 'galio-framework';
import { cloudinary } from 'cloudinary-react';

export default function ImagePickerExample({ onImagePicked, defaultImage }) {
  const { width, height } = Dimensions.get('screen');
  const [image, setImage] = useState(null);

  useEffect(() => {
    (async () => {
      if (Platform.OS !== 'web') {
        const { status } = await ImagePicker.requestCameraRollPermissionsAsync();
        if (status !== 'granted') {
          alert('Sorry, we need camera roll permissions to make this work!');
        }
      }
    })();
  }, []);

  const pickImage = async () => {
    let result = await ImagePicker.launchImageLibraryAsync({
      mediaTypes: ImagePicker.MediaTypeOptions.All,
      allowsEditing: true,
      aspect: [4, 3],
      quality: 1,
    });

    if (!result.cancelled) {
      let file = {
        uri: result.uri,
        type: `test/${result.uri.split('.').pop()}`,
        name: result.uri,
      };

      let apiUrl = 'https://api.cloudinary.com/v1_1/mixosteam/image/upload';

      const formData = new FormData();
      formData.append('file', file);
      formData.append('upload_preset', 'zc9ttas7');
      formData.append('cloud_name', 'mixosteam');

      await fetch(apiUrl, {
        body: formData,
        method: 'POST',
      })
        .then(async (r) => await r.json())
        .then((dato) => {
          console.log('TOY AQUIIII');
          setImage(dato.secure_url);
          onImagePicked({ uri: dato.secure_url });
        })
        .catch((error) => {
          console.log(error);
        });
    }
  };

  const styles = StyleSheet.create({
    smallButton: {
      width: 75,
      height: 28,
    },
    Button: {
      width: width / 2.5,
      height: height / 20,
      shadowOpacity: 0,
    },
  });

  return (
    <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center',shadowOpacity: 0}}>
      <Button style={styles.Button} onPress={pickImage}>
        Subir imagen
      </Button>
      {image && (
        <Image source={{ uri: image }} style={{ width: 200, height: 200, marginTop: 20 }} />
      )}
      {defaultImage && !image && (
        <Image source={{ uri: defaultImage }} style={{ width: 200, height: 200, marginTop: 20 }} />
      )}
    </View>
  );
}
