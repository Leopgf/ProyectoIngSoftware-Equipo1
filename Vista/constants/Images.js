// local imgs
const Onboarding = require('../assets/imgs/FFondo.jpg');
const Logo = require('../assets/imgs/logop.png');
const ProfileBackground = require('../assets/imgs/basePerfil.jpg');
const RegisterBackground = require('../assets/imgs/fondoInicio.jpg');
const Pro = require('../assets/imgs/bg.png');
const NowLogo = require('../assets/imgs/logo.png');
const iOSLogo = require('../assets/imgs/ios.png');
const androidLogo = require('../assets/imgs/android.png');
const ProfilePicture = require('../assets/imgs/profile-img.jpg');
const Conversor = require('../assets/imgs/conversor.jpg');
const AddReceta = require('../assets/imgs/addReceta.jpg');
const CreativeTimLogo = require('../assets/imgs/creative-tim-white-slim2.png');
const InvisionLogo = require('../assets/imgs/invision-white-slim.png');

const Viewed = [
  require('../assets/imgs/project5.jpg'),
  require('../assets/imgs/project7.jpg'),
  require('../assets/imgs/project6.jpg'),
  require('../assets/imgs/project4.jpg'),
  require('../assets/imgs/project21.jpg'),
  require('../assets/imgs/project24.jpg')
];

const Products = {
  text: 'View article',
  path: require('../assets/imgs/bg15.jpg')
};

export default {
  Onboarding,
  Logo,
  ProfileBackground,
  Conversor,
  ProfilePicture,
  RegisterBackground,
  AddReceta,
  Viewed,
  Pro,
  Products,
  NowLogo: NowLogo
};
