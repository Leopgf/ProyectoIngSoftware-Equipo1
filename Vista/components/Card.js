//IMPORT
import React from 'react';
import { withNavigation } from '@react-navigation/compat';
import PropTypes from 'prop-types';
import { Button as GaButton } from 'galio-framework';
import { StyleSheet, Image, TouchableWithoutFeedback } from 'react-native';
import { Block, Text, theme } from 'galio-framework';
import { nowTheme } from '../constants';

import * as firebase from 'firebase';

//CLASE DE CARD
class Card extends React.Component {
   render() {
    const {
      navigation,
      item,
      horizontal,
      full,
      style,
      ctaColor,
      imageStyle,
      ctaRight,
      titleStyle,
      params,
    } = this.props;

    const imageStyles = [full ? styles.fullImage : styles.horizontalImage, imageStyle];
    const titleStyles = [styles.cardTitle, titleStyle];
    const cardContainer = [styles.card, styles.shadow, style];
    const imgContainer = [
      styles.imageContainer,
      horizontal ? styles.horizontalStyles : styles.verticalStyles,
      styles.shadow,
    ];

    var user = '';

    if(item.userID){
      firebase.firestore().collection('Usuarios').where('usuarioID', '==', item.userID).get().then((usuarios) => {

        usuarios.forEach((usuario) => {
          user = usuario.data().usuario;
        });
      });
    }

    if(item.nombre){
      return (
        <Block row={horizontal} card flex style={cardContainer}>
          <TouchableWithoutFeedback onPress={() => navigation.navigate('DetallesReceta', params)}>
            <Block flex style={imgContainer}>
              <Image
                resizeMode="cover"
                source={item.imagen && { uri: item.imagen }}
                style={imageStyles}
              />
            </Block>
          </TouchableWithoutFeedback>
  
          <TouchableWithoutFeedback onPress={() => navigation.navigate('DetallesReceta', params)}>
            <Block flex space="between" style={styles.cardDescription}>
              <Block flex>
                <Block flex style={{ justifyContent: 'center' }}>
                  {item.nombre && !horizontal ? (
                    <Text size={24} style={styles.cardTitle} color={'#e63746'}>
                      {/*TITULO DE LA RECETA*/}
                      {item.nombre}
                    </Text>
                  ) : (
                    <Text size={24} style={styles.cardTitle2} color={'#e63746'}>
                      {/*TITULO DE LA RECETA*/}
                      {item.nombre}
                    </Text>
                  )}
                </Block>
                {item.subtitle ? (
                  <Block flex center>
                    <Text
                      style={{ fontFamily: 'montserrat-regular' }}
                      size={32}
                      color={nowTheme.COLORS.BLACK}
                    >
                      {item.subtitle}
                    </Text>
                  </Block>
                ) : null}
                {/*DESCRIPCION RECETA*/}
                {item.descripcion && !horizontal ? (
                  <Block flex center>
                    <Text
                      style={{ fontFamily: 'montserrat-regular', textAlign: 'left', padding: 15 }}
                      size={14}
                      color={'#0f1e2e'}
                    >
                      {item.descripcion}
                    </Text>
                  </Block>
                ) : null}
                {item.body ? (
                  <Block flex left>
                    <Text
                      style={{ fontFamily: 'montserrat-regular' }}
                      size={12}
                      color={nowTheme.COLORS.TEXT}
                    >
                      {item.body}
                    </Text>
                  </Block>
                ) : null}
              </Block>
              {ctaRight ? (
                <Block right={ctaRight ? true : false}>
                  <Text
                    style={styles.articleButton}
                    size={12}
                    muted={!ctaColor}
                    color={ctaColor || nowTheme.COLORS.ACTIVE}
                    bold
                  >
                    {item.cta}
                  </Text>
                </Block>
              ) : null}
            </Block>
          </TouchableWithoutFeedback>
        </Block>
      );
    }else{
        return (
          <Block row={horizontal} card flex style={cardContainer}>
            <Block flex style={imgContainer}>
              <Image
                resizeMode="cover"
                source={item.imagen && { uri: item.imagen }}
                style={imageStyles}
              />
            </Block>
  
            <Block flex space="between" style={styles.cardDescription}>
              <Block flex>
                <Block flex style={{ justifyContent: 'center' }}>
                  {item.titulo && !horizontal ? (
                    <Text size={24} style={styles.cardTitle} color={'#e63746'}>
                      {/*TITULO DE LA REVIEW*/}
                      {item.titulo}
                    </Text>
                  ) : (
                    <Text size={24} style={styles.cardTitle2} color={'#e63746'}>
                      {/*TITULO DE LA REVIEW*/}
                      {item.titulo}
                    </Text>
                  )}
                </Block>
                {/*DESCRIPCION REVIEW*/}
                {item.mensaje ? (
                  <Block flex center>
                    <Text
                      style={{ fontFamily: 'montserrat-regular', textAlign: 'left', padding: 15 }}
                      size={14}
                      color={'#0f1e2e'}
                    >
                      {item.mensaje}
                    </Text>
                  </Block>
                ) : null}
                {item.userID && item.valoracion ? (
                  <Block flex left>
                    <Text
                      style={{ fontFamily: 'montserrat-regular' }}
                      size={12}
                      color={nowTheme.COLORS.TEXT}
                    >
                      {item.userID}
                    </Text>
                    <Text
                      style={{ fontFamily: 'montserrat-regular' }}
                      size={12}
                      color={nowTheme.COLORS.TEXT}
                    >
                      {item.valoracion}
                    </Text>
                    <GaButton
                          round
                          onlyIcon
                          shadowless
                          icon="star"
                          iconFamily="Font-Awesome"
                          iconColor={'#E63746'}
                          iconSize={nowTheme.SIZES.BASE * 1.4}
                          color={'#FFFFFF'}
                          style={[styles.social, styles.shadow]}
                        />
                  </Block>
                ) : null}
              </Block>
              {ctaRight ? (
                <Block right={ctaRight ? true : false}>
                  <Text
                    style={styles.articleButton}
                    size={12}
                    muted={!ctaColor}
                    color={ctaColor || nowTheme.COLORS.ACTIVE}
                    bold
                  >
                    {item.cta}
                  </Text>
                </Block>
              ) : null}
            </Block>
          </Block>
        );
    }
  }
}

Card.propTypes = {
  item: PropTypes.object,
  horizontal: PropTypes.bool,
  full: PropTypes.bool,
  ctaColor: PropTypes.string,
  imageStyle: PropTypes.any,
  ctaRight: PropTypes.bool,
  titleStyle: PropTypes.any,
  textBodyStyle: PropTypes.any,
};

const styles = StyleSheet.create({
  card: {
    backgroundColor: '#F2F2F3',
    marginVertical: theme.SIZES.BASE,
    borderWidth: 0,
    minHeight: 114,
    marginBottom: 4,
  },
  cardTitle: {
    paddingHorizontal: 9,
    paddingTop: 7,
    paddingBottom: 2,
    fontSize: 24,
    textAlign: 'center',
    fontWeight: 'bold',
  },
  cardTitle2: {
    paddingHorizontal: 9,
    paddingTop: 7,
    paddingBottom: 2,
    fontSize: 24,
    textAlign: 'center',
    fontWeight: 'bold',
    color: '#0f1e2e'
  },
  cardDescription: {
    // padding: theme.SIZES.BASE / 2
  },
  imageContainer: {
    borderRadius: 4,
    elevation: 1,
    overflow: 'hidden',
  },
  image: {
    // borderRadius: 3,
  },
  horizontalImage: {
    height: 200,
    width: 'auto',
  },
  horizontalStyles: {
    borderTopRightRadius: 0,
    borderBottomRightRadius: 0,
  },
  verticalStyles: {
    borderBottomRightRadius: 0,
    borderBottomLeftRadius: 0,
  },
  fullImage: {
    height: 215,
  },
  shadow: {
    shadowColor: '#8898AA',
    shadowOffset: { width: 0, height: 1 },
    shadowRadius: 6,
    shadowOpacity: 0.1,
    elevation: 2,
  },
  articleButton: {
    fontFamily: 'montserrat-bold',
    paddingHorizontal: 9,
    paddingVertical: 7,
  },
});

export default withNavigation(Card);
