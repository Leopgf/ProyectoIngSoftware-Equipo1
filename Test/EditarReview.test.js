import React from 'react';
import renderer from 'react-test-renderer';

import {
  StyleSheet,
  ImageBackground,
  Dimensions,
  TouchableWithoutFeedback,
  Keyboard,
} from 'react-native';
import LoadingView from 'react-native-loading-view';
import { Block, Text, Button as GaButton, theme } from 'galio-framework';

import { Button, Icon, Input } from '../Vista/components';
import { Images, nowTheme } from '../Vista/constants';
import ImagePickerExample from '../Vista/components/ImagePicker';
import { ScrollView } from 'react-native-gesture-handler';
import ModalDropdown from 'react-native-modal-dropdown';
const { width, height } = Dimensions.get('screen');

const DismissKeyboard = ({ children }) => (
  <TouchableWithoutFeedback onPress={() => Keyboard.dismiss()}>{children}</TouchableWithoutFeedback>
);

describe('<EditReview />', () => {
  //
  it('renders correctly', async () => {
    const tree = await renderer
      .create(
        <Block middle>
          <Text
            style={{
              fontFamily: 'montserrat-bold',
              textAlign: 'center',
              fontWeight: '500',
              marginTop: 10,
            }}
            color="#e63746"
            size={24}
          >
            Editar Review
          </Text>
          <Text
            style={{
              fontFamily: 'montserrat-bold',
              textAlign: 'center',
              fontWeight: '500',
              marginTop: 10,
            }}
            color="#0f1e2e"
            size={14}
          >
            Edita tu review de la receta
          </Text>
        </Block>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders correctly', async () => {
    const tree = await renderer
      .create(
        <Block flex space="between">
          {/* NO SE POR QUE NO SE VE AAAAAAAAAAAAA */}
          <ImagePickerExample />
          <Input
            placeholder="Titulo"
            style={styles.inputs}
            iconContent={<Icon size={18} name="message" family="ArgonExtra" />}
          />
          <Input
            placeholder="Mensaje"
            style={styles.inputs}
            iconContent={<Icon size={18} name="message" family="ArgonExtra" />}
          />
          <Text>Valoraci√≥n:</Text>
          <Button
            style={{
              fontFamily: 'montserrat-bold',
              borderRadius: nowTheme.SIZES.BASE * 1.5,
              width: 200,
              marginBottom: 30,
              marginTop: 30,
            }}
            color="primary"
            round
          >
            ACTUALIZAR
          </Button>
        </Block>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

const styles = StyleSheet.create({
  imageBackgroundContainer: {
    width: width,
    height: height,
    padding: 0,
    zIndex: 1,
  },
  imageBackground: {
    width: width,
    height: height,
  },
  registerContainer: {
    marginTop: 55,
    width: width * 0.9,
    height: height < 812 ? height * 0.8 : height * 0.8,
    backgroundColor: nowTheme.COLORS.WHITE,
    borderRadius: 4,
    shadowColor: nowTheme.COLORS.BLACK,
    shadowOffset: {
      width: 0,
      height: 4,
    },
    shadowRadius: 8,
    shadowOpacity: 0.1,
    elevation: 1,
    overflow: 'hidden',
  },
  dropdown: {
    alignSelf: 'center',
    width: width / 4,
    height: height / 22,
    marginTop: 10,
    alignItems: 'center',
    right: 8,
    borderWidth: 0,
    borderRadius: 15,
    backgroundColor: '#e63746',
  },

  dropdownOption: {
    width: width / 5,
    alignItems: 'center',
    borderColor: '#e63746',
    borderRadius: 10,
    borderWidth: 2,
    borderRadius: 3,
  },
  dropdownText: {
    marginVertical: 10,
    marginHorizontal: 6,
    fontSize: 18,
    color: 'white',
    textAlign: 'center',
    textAlignVertical: 'center',
  },
  socialConnect: {
    backgroundColor: nowTheme.COLORS.WHITE,
    // borderBottomWidth: StyleSheet.hairlineWidth,
    // borderColor: "rgba(136, 152, 170, 0.3)"
  },
  socialButtons: {
    width: 120,
    height: 40,
    backgroundColor: '#fff',
    shadowColor: nowTheme.COLORS.BLACK,
    shadowOffset: {
      width: 0,
      height: 4,
    },
    shadowRadius: 8,
    shadowOpacity: 0.1,
    elevation: 1,
  },
  socialTextButtons: {
    color: nowTheme.COLORS.PRIMARY,
    fontWeight: '800',
    fontSize: 14,
  },
  inputIcons: {
    marginRight: 12,
    color: nowTheme.COLORS.ICON_INPUT,
  },
  inputs: {
    width: width * 0.65,
    borderWidth: 1,
    borderColor: '#E3E3E3',
    borderRadius: 21.5,
    alignSelf: 'center',
  },
  passwordCheck: {
    paddingLeft: 2,
    paddingTop: 6,
    paddingBottom: 15,
  },
  createButton: {
    width: width * 0.5,
    marginTop: 25,
    marginBottom: 40,
  },
  social: {
    width: theme.SIZES.BASE * 3.5,
    height: theme.SIZES.BASE * 3.5,
    borderRadius: theme.SIZES.BASE * 1.75,
    justifyContent: 'center',
    marginHorizontal: 10,
  },
});
