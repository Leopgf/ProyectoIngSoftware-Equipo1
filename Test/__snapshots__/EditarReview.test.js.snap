// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditReview /> renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "column",
      },
      false,
      false,
      false,
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      false,
      false,
      false,
      false,
      null,
      false,
      false,
      false,
      null,
      null,
      null,
      undefined,
    ]
  }
  theme={
    Object {
      "COLORS": Object {
        "BLACK": "#000000",
        "BLOCK": "#808080",
        "DRIBBBLE": "#EA4C89",
        "ERROR": "#FE2472",
        "FACEBOOK": "#3B5998",
        "GREY": "#898989",
        "ICON": "#000000",
        "INFO": "#1232FF",
        "INPUT": "#808080",
        "MUTED": "#9FA5AA",
        "NAVBAR": "#F9F9F9",
        "NEUTRAL": "rgba(255,255,255, 0.65)",
        "PLACEHOLDER": "#9FA5AA",
        "PRIMARY": "#B23AFC",
        "SUCCESS": "#45DF31",
        "THEME": "#B23AFC",
        "TRANSPARENT": "transparent",
        "TWITTER": "#5BC0DE",
        "WARNING": "#FF9C09",
        "WHITE": "#FFFFFF",
      },
      "SIZES": Object {
        "ANDROID_ELEVATION": 1,
        "BASE": 16,
        "BLOCK_SHADOW_OPACITY": 0.15,
        "BLOCK_SHADOW_RADIUS": 8,
        "BORDER_RADIUS": 6,
        "BORDER_WIDTH": 0.8,
        "BUTTON_HEIGHT": 44,
        "BUTTON_SHADOW_RADIUS": 10,
        "BUTTON_WIDTH": 144,
        "CARD_AVATAR_HEIGHT": 40,
        "CARD_AVATAR_RADIUS": 20,
        "CARD_AVATAR_WIDTH": 40,
        "CARD_BORDER_RADIUS": 6.4,
        "CARD_BORDER_WIDTH": 0.8,
        "CARD_FOOTER_HORIZONTAL": 12,
        "CARD_FOOTER_VERTICAL": 12,
        "CARD_IMAGE_HEIGHT": 200,
        "CARD_MARGIN_VERTICAL": 14,
        "CARD_ROUND": 3,
        "CARD_ROUNDED": 8,
        "CARD_WIDTH": 718,
        "CHECKBOX_HEIGHT": 20,
        "CHECKBOX_WIDTH": 20,
        "FONT": 16,
        "ICON": 16,
        "INPUT_BORDER_RADIUS": 8,
        "INPUT_BORDER_WIDTH": 0.8,
        "INPUT_HEIGHT": 44,
        "INPUT_HELP_TEXT": 12.8,
        "INPUT_HORIZONTAL": 16,
        "INPUT_LABEL_BOTTOM": 4,
        "INPUT_LABEL_TEXT": 14.4,
        "INPUT_ROUNDED": 27.2,
        "INPUT_TEXT": 14,
        "NAVBAR_HEIGHT": 66,
        "NAVBAR_LEFT_FLEX": 0.5,
        "NAVBAR_LEFT_HEIGHT": 93.38000000000001,
        "NAVBAR_LEFT_MARGIN": 16,
        "NAVBAR_RIGHT_FLEX": 0.5,
        "NAVBAR_RIGHT_HEIGHT": 93.38000000000001,
        "NAVBAR_RIGHT_MARGIN": 16,
        "NAVBAR_TITLE_FLEX": 2,
        "NAVBAR_TITLE_HEIGHT": 93.38000000000001,
        "NAVBAR_TITLE_TEXT": 14,
        "NAVBAR_VERTICAL": 16,
        "OPACITY": 0.8,
        "RADIO_HEIGHT": 24,
        "RADIO_THICKNESS": 2,
        "RADIO_WIDTH": 24,
        "THUMB_SIZE": 25,
        "TRACK_SIZE": 4,
      },
    }
  }
>
  <Text
    style={
      Array [
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        undefined,
        Object {
          "fontSize": 24,
        },
        Object {
          "color": "#e63746",
        },
        false,
        false,
        undefined,
        Object {
          "fontFamily": "montserrat-bold",
          "fontWeight": "500",
          "marginTop": 10,
          "textAlign": "center",
        },
      ]
    }
  >
    Editar Review
  </Text>
  <Text
    style={
      Array [
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        undefined,
        Object {
          "fontSize": 14,
        },
        Object {
          "color": "#0f1e2e",
        },
        false,
        false,
        undefined,
        Object {
          "fontFamily": "montserrat-bold",
          "fontWeight": "500",
          "marginTop": 10,
          "textAlign": "center",
        },
      ]
    }
  >
    Edita tu review de la receta
  </Text>
</View>
`;

exports[`<EditReview /> renders correctly 2`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "column",
      },
      false,
      Object {
        "flex": 1,
      },
      false,
      false,
      false,
      false,
      false,
      false,
      Object {
        "justifyContent": "space-between",
      },
      false,
      false,
      false,
      null,
      null,
      null,
      undefined,
    ]
  }
  theme={
    Object {
      "COLORS": Object {
        "BLACK": "#000000",
        "BLOCK": "#808080",
        "DRIBBBLE": "#EA4C89",
        "ERROR": "#FE2472",
        "FACEBOOK": "#3B5998",
        "GREY": "#898989",
        "ICON": "#000000",
        "INFO": "#1232FF",
        "INPUT": "#808080",
        "MUTED": "#9FA5AA",
        "NAVBAR": "#F9F9F9",
        "NEUTRAL": "rgba(255,255,255, 0.65)",
        "PLACEHOLDER": "#9FA5AA",
        "PRIMARY": "#B23AFC",
        "SUCCESS": "#45DF31",
        "THEME": "#B23AFC",
        "TRANSPARENT": "transparent",
        "TWITTER": "#5BC0DE",
        "WARNING": "#FF9C09",
        "WHITE": "#FFFFFF",
      },
      "SIZES": Object {
        "ANDROID_ELEVATION": 1,
        "BASE": 16,
        "BLOCK_SHADOW_OPACITY": 0.15,
        "BLOCK_SHADOW_RADIUS": 8,
        "BORDER_RADIUS": 6,
        "BORDER_WIDTH": 0.8,
        "BUTTON_HEIGHT": 44,
        "BUTTON_SHADOW_RADIUS": 10,
        "BUTTON_WIDTH": 144,
        "CARD_AVATAR_HEIGHT": 40,
        "CARD_AVATAR_RADIUS": 20,
        "CARD_AVATAR_WIDTH": 40,
        "CARD_BORDER_RADIUS": 6.4,
        "CARD_BORDER_WIDTH": 0.8,
        "CARD_FOOTER_HORIZONTAL": 12,
        "CARD_FOOTER_VERTICAL": 12,
        "CARD_IMAGE_HEIGHT": 200,
        "CARD_MARGIN_VERTICAL": 14,
        "CARD_ROUND": 3,
        "CARD_ROUNDED": 8,
        "CARD_WIDTH": 718,
        "CHECKBOX_HEIGHT": 20,
        "CHECKBOX_WIDTH": 20,
        "FONT": 16,
        "ICON": 16,
        "INPUT_BORDER_RADIUS": 8,
        "INPUT_BORDER_WIDTH": 0.8,
        "INPUT_HEIGHT": 44,
        "INPUT_HELP_TEXT": 12.8,
        "INPUT_HORIZONTAL": 16,
        "INPUT_LABEL_BOTTOM": 4,
        "INPUT_LABEL_TEXT": 14.4,
        "INPUT_ROUNDED": 27.2,
        "INPUT_TEXT": 14,
        "NAVBAR_HEIGHT": 66,
        "NAVBAR_LEFT_FLEX": 0.5,
        "NAVBAR_LEFT_HEIGHT": 93.38000000000001,
        "NAVBAR_LEFT_MARGIN": 16,
        "NAVBAR_RIGHT_FLEX": 0.5,
        "NAVBAR_RIGHT_HEIGHT": 93.38000000000001,
        "NAVBAR_RIGHT_MARGIN": 16,
        "NAVBAR_TITLE_FLEX": 2,
        "NAVBAR_TITLE_HEIGHT": 93.38000000000001,
        "NAVBAR_TITLE_TEXT": 14,
        "NAVBAR_VERTICAL": 16,
        "OPACITY": 0.8,
        "RADIO_HEIGHT": 24,
        "RADIO_THICKNESS": 2,
        "RADIO_WIDTH": 24,
        "THUMB_SIZE": 25,
        "TRACK_SIZE": 4,
      },
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "shadowOpacity": 0,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#B23AFC",
          "borderRadius": 3,
          "elevation": 1,
          "height": 66.7,
          "justifyContent": "center",
          "opacity": 1,
          "shadowColor": "#B23AFC",
          "shadowOffset": Object {
            "height": 4,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 10,
          "width": 300,
          "zIndex": 2,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 16,
            },
            undefined,
          ]
        }
      >
        Subir imagen
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignContent": "center",
        "marginVertical": 8,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#FFFFFF",
            "borderColor": "#808080",
            "borderRadius": 8,
            "borderWidth": 0.8,
            "height": 44,
            "paddingHorizontal": 16,
            "width": "100%",
          },
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          },
          null,
          false,
          false,
          Object {
            "alignSelf": "center",
            "borderColor": "#E3E3E3",
            "borderRadius": 21.5,
            "borderWidth": 1,
            "width": 487.5,
          },
        ]
      }
    >
      <TextInput
        allowFontScaling={true}
        error={false}
        keyboardType="default"
        onRef={null}
        placeholder="Titulo"
        placeholderTextColor="#8898AA"
        primary={false}
        rejectResponderTermination={true}
        secureTextEntry={false}
        shadowless={false}
        style={
          Array [
            Object {
              "flex": 1,
            },
            false,
            Object {
              "color": "#808080",
              "fontSize": 14,
              "textDecorationColor": "transparent",
              "textShadowColor": "transparent",
            },
            Object {
              "color": "#0f1e2e",
            },
          ]
        }
        success={false}
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignContent": "center",
        "marginVertical": 8,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#FFFFFF",
            "borderColor": "#808080",
            "borderRadius": 8,
            "borderWidth": 0.8,
            "height": 44,
            "paddingHorizontal": 16,
            "width": "100%",
          },
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          },
          null,
          false,
          false,
          Object {
            "alignSelf": "center",
            "borderColor": "#E3E3E3",
            "borderRadius": 21.5,
            "borderWidth": 1,
            "width": 487.5,
          },
        ]
      }
    >
      <TextInput
        allowFontScaling={true}
        error={false}
        keyboardType="default"
        onRef={null}
        placeholder="Mensaje"
        placeholderTextColor="#8898AA"
        primary={false}
        rejectResponderTermination={true}
        secureTextEntry={false}
        shadowless={false}
        style={
          Array [
            Object {
              "flex": 1,
            },
            false,
            Object {
              "color": "#808080",
              "fontSize": 14,
              "textDecorationColor": "transparent",
              "textShadowColor": "transparent",
            },
            Object {
              "color": "#0f1e2e",
            },
          ]
        }
        success={false}
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
  <Text
    style={
      Array [
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        undefined,
        0,
        null,
        false,
        false,
        undefined,
        null,
      ]
    }
  >
    Valoración:
  </Text>
  <View
    accessible={true}
    focusable={false}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#B23AFC",
        "borderRadius": 24,
        "elevation": 1,
        "fontFamily": "montserrat-bold",
        "height": 44,
        "justifyContent": "center",
        "marginBottom": 30,
        "marginTop": 30,
        "opacity": 1,
        "shadowColor": "#B23AFC",
        "width": 200,
        "zIndex": 2,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 16,
          },
          Object {
            "fontSize": 12,
            "fontWeight": "700",
          },
        ]
      }
    >
      ACTUALIZAR
    </Text>
  </View>
</View>
`;
